File: csharp/src/JokeApi/Services/JokeService.cs
using JokeApi.Models;
using JokeApi.Repositories;

namespace JokeApi.Services;

public class JokeService
{
    private readonly IJokeRepository _repo;

    public JokeService(IJokeRepository repo)
    {
        _repo = repo;
    }

    public Task<IEnumerable<Joke>> GetAllAsync() => _repo.GetAllAsync();
    public Task<Joke?> GetByIdAsync(int id) => _repo.GetByIdAsync(id);
    public Task<Joke> CreateAsync(JokeCreate c) => _repo.CreateAsync(c);
    public Task<bool> UpdateAsync(int id, Joke j) => _repo.UpdateAsync(id, j);
    public Task<bool> DeleteAsync(int id) => _repo.DeleteAsync(id);
    public Task<Joke?> GetRandomAsync(string? category) => _repo.GetRandomAsync(category);
    public Task<IEnumerable<Joke>> GetBySourceAsync(string source) => _repo.GetBySourceAsync(source);
    public Task<int> IncrementLolAsync(int id) => _repo.IncrementLolAsync(id);
    public Task<int> IncrementGroanAsync(int id) => _repo.IncrementGroanAsync(id);
}

File: kotlin/app/src/main/kotlin/com/example/services/JokeService.kt
package com.example.services

import com.example.models.Joke
import com.example.models.JokeCreate
import com.example.repositories.JokeRepository

class JokeService(private val repo: JokeRepository) {
    suspend fun getAll(): List<Joke> = repo.getAll()
    suspend fun getById(id: Int): Joke? = repo.getById(id)
    suspend fun create(create: JokeCreate): Joke = repo.create(create)
    suspend fun update(id: Int, joke: Joke): Boolean = repo.update(id, joke)
    suspend fun delete(id: Int): Boolean = repo.delete(id)
    suspend fun getRandom(category: String?): Joke? = repo.getRandom(category)
    suspend fun getBySource(source: String): List<Joke> = repo.getBySource(source)
    suspend fun incrementLol(id: Int): Int = repo.incrementLol(id)
    suspend fun incrementGroan(id: Int): Int = repo.incrementGroan(id)
}
